File: frontend/modules/trigger/application/trigger-finder.ts
import { TriggerRepository } from '../domain/trigger-repository'
import { Trigger } from '../domain/trigger'

export class TriggerFinder {

  constructor(private repository: TriggerRepository) { }

  async findByDiscordServerId(discordServerId: string): Promise<Trigger[]> {
    return await this.repository.findByDiscordServerId(discordServerId)
  }

}

File: context/src/main/kotlin/com/usadapekora/context/trigger/application/TriggerFinder.kt
package com.usadapekora.context.trigger.application

import com.usadapekora.context.trigger.domain.Trigger
import com.usadapekora.context.trigger.domain.TriggerException
import com.usadapekora.context.trigger.domain.TriggerMatcher
import com.usadapekora.context.trigger.domain.TriggerRepository

class TriggerFinder(
    private val repository: TriggerRepository,
    private val matcher: TriggerMatcher
) {
    suspend fun findByInput(input: String, discordServerId: String): Trigger {
        val triggers = repository.findByDiscordServer(Trigger.TriggerDiscordGuildId(discordServerId))
        return matcher.matchInput(input, triggers) ?: throw TriggerException.NotFound()
    }
}
